labeling/internal/python.go

```diff
@@ -107,22 +109,67 @@ func fileContainsString(c codebase.Codebase, filePath string, str string) bool {
 }
 
 func getPythonVersion(c codebase.Codebase) string {
-	versionRegex := regexp.MustCompile(`[0-9.]+`)
 
 	versionFilePath, _ := c.FindFile(".python-version")
-	if versionFilePath == "" {
-		return ""
-	}
+	if versionFilePath != "" {
+		file, err := c.ReadFile(versionFilePath)
+		if err != nil {
+			log.Println("Unable to read file .python-version. Attempting to determine python version another way... ", err)
+		} else {
+			versionRegex := regexp.MustCompile(`[0-9.]+`)
+			pythonVersion := versionRegex.FindString(string(file))
+			if pythonVersion != "" {
+				log.Println("Unable to parse file .python-version. Attempting to determine python version another way...")
 
-	file, err := c.ReadFile(versionFilePath)
-	if err != nil {
-		return ""
+			}
+			log.Println("Found Python version in .python-version file: ", pythonVersion)
+			return pythonVersion
+		}
 	}
 
-	version := versionRegex.FindString(string(file))
-	if version != "" {
-		return version
+	pyprojectFilePath, _ := c.FindFile("pyproject.toml")
+	if pyprojectFilePath != "" {
+		file, err := c.ReadFile(pyprojectFilePath)
+		if err != nil {
+			log.Println("Unable to read file pyproject.toml. Attempting to determine python version another way... ", err)
+		} else {
+			tree, err := toml.LoadBytes(file)
+			if err != nil {
+				log.Println("Unable to read file pyproject.toml. Attempting to determine python version another way... ", err)
+			} else {
+				pythonVersion := tree.Get("tool.poetry.dependencies.python")
+				if pythonVersion == nil || pythonVersion.(string) == "" {
+					log.Println("Error parsing pyproject.toml, python version is nil")
+				} else {
```

Comment:
~ Nit: perhaps some helper functions could reduce the level of nesting here - e.g. those functions can return early on error so that "else" isn't needed.